packages/react-api/src/hooks/gpu-filter-utils.js

```diff
@@ -0,0 +1,88 @@
+import { _buildFeatureFilter, _FilterTypes } from '@carto/react-core/';
+
+// Don't change this value to maintain compatibility with builder
+export const MAX_GPU_FILTERS = 4;
+
+function getFiltersWithoutTimeType(filters) {
+  const filtersWithoutTimeType = {};
+  let timeFilter = {};
+
+  Object.entries(filters).forEach(([column, columnData]) => {
+    Object.entries(columnData).forEach(([type, typeData]) => {
+      if (type !== _FilterTypes.TIME) {
+        filtersWithoutTimeType[column] = { [type]: typeData };
+      } else {
+        timeFilter = {
+          column,
+          values: typeData.values,
+          type
+        };
+      }
+    });
+  });
+  return {
+    filtersWithoutTimeType,
+    timeFilter
+  };
+}
+
+function getFilterRange(timeFilter) {
+  const result = Array(MAX_GPU_FILTERS).fill([0, 0]);
+  // According to getFilterValue all filters are resolved as 0 or 1 in the first position of the array
+  // except the time filter value that is resolved with the real value of the feature in the second position of the array
+  result[0] = [1, 1];
+  if (Object.keys(timeFilter).length > 0) {
+    result[1] = timeFilter.values[0];
+  }
+  return result;
+}
+
+function getFilterValueUpdateTriggers(filtersWithoutTimeType, timeFilter) {
+  const result = { ...filtersWithoutTimeType };
+
+  // We don't want to change the layer UpdateTriggers evety time that the time filter changes
+  // because this filter is changed by the time series widget during its animation
+  // so we remove the the time filter value from the UpdateTriggers
+  if (Object.keys(timeFilter).length > 0) {
+    result[timeFilter.column] = {
+      ...result[timeFilter.column],
+      [timeFilter.type]: {}
+    };
+  }
+  return JSON.stringify(result);
+}
+
+function getFilterValue(filtersWithoutTimeType, timeFilter) {
+  const result = Array(MAX_GPU_FILTERS).fill(0);
+  const hastimeFilterValues = Object.keys(timeFilter).length > 0;
+
+  // We evaluate all filters except the time filter using _buildFeatureFilter function.
+  // For the time filter, we return the value of the feature and we will change the getFilterRange result
+  // every time this filter changes
+  return (feature) => {
+    result[0] = _buildFeatureFilter({ filters: filtersWithoutTimeType, type: 'number' })(
+      feature
+    );
+
+    if (hastimeFilterValues) {
+      const f = feature.properties || feature;
+      result[1] = f[timeFilter.column];
+    }
+    return result;
+  };
+}
+
+// The deck.gl DataFilterExtension accepts up to 4 values to filter.
+// We're going to use the first value for all filter except the time filter
+// that will be managed by the second value of the DataFilterExtension
+export function getGpuFilter(filters = {}) {
```

Comment:
Podríamos ponerle `getDataFilterExtensionProps` y sacamos tb de aquí el extensions y el updateTriggers para inyectarlo en el return del `useCartoLayerProps` como se hace con la constante `props`? No soy demasiado fan de llamarlo `getGpuFilter`, no se está haciendo dentro de la función nada en GPU, solo se están devolviendo props del DataFilterExtension. Lo mismo haría con el nombre del archivo y los tests, no me convence llamarlo GPUFilter, parece que es algo paralelo al DataFilterExtension que es un nombre ya conocido por quien usa deck.gl. Creo que así queda mucho más claro.

Reply:
I like the name `getDataFilterExtensionProps`

Reply:
De hecho he sacado los nombres del kepler, ellos tienes un fichero que se llama `gpu-filter-utils` tal cual que se utiliza para esto mismo.

Reply:
I think it's because DataFilterExtension wasn't an extension in the past and had different names like DataFilter or maybe GPUFilter. Idk. But In C4R I would call it `getDataFilterExtensionsProps` and I guess that the filename should be in camelCase, right?